sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -894,7 +894,7 @@ object SparkSession extends Logging {
      *
      * @since 2.2.0
      */
-    def withExtensions(f: SparkSessionExtensions => Unit): Builder = synchronized {
+    def withExtensions(f: SparkSessionExtensionsProvider): Builder = synchronized {
```

Comment:
Hm, I think this breaks binary compatibility.

Reply:
I see